### ISSUE TITLE
This addon crashes Firefox since the last TST update.

### ISSUE THREAD

**Issue by jarredou** [piroor/tst-lock-tree-collapsed#14]
Since the last TST update I have to disable this addon because it crashes Firefox each time just after TST's sidebar is loaded.
CPU and Memory consumption are going crazy ( Firefox taking +1GB memory each 5 seconds, until crash).

When disabled, everything is ok.

**Comment by jarredou** [comment]
This bug is also occuring when I start Firefox with tst-lock-tree-collapsed disabled and enable it only after the TST sidebar is loaded.

**Comment by jarredou** [comment]
Ok, I've seen the warning in the changelog of TST v3.8.7 about compatibility problems that could occur with helper addons.
I hope this one will be fixed soon, I can't live without it ! :)

I'm staying with the previous version of TST until then.

**Comment by jarredou** [comment]
Will you fix this helper or it's dead now ?

**Comment by piroor** [comment]
Hmm, this problem doesn't happen on my environment Firefox 91 on Windows 10...

**Comment by jarredou** [comment]
It is happening again, it only happens when you have many (+100~200) tabs opened (even if they are all discarded).
Same thing than last time, RAM goes crazy until system freeze/crash.
This is only happening when [tst-lock-tree-collapsed](https://github.com/piroor/tst-lock-tree-collapsed) is enabled. No RAM problem when it's disabled.

EDIT : This is also happening when Firefox is already started and if you load TST's sidebar after.

**Comment by jarredou** [comment]
Still happening...

**Comment by dsl101** [comment]
I just installed this add-on in relation to [this](https://github.com/piroor/treestyletab/issues/3287), and also see some strange memory-related behaviour. FF was very sluggish to open from cold (like ≈30 seconds until responsive), and the task manager memory profile looks like this.

![image](https://user-images.githubusercontent.com/900542/221602362-bfe9a506-ec25-448b-bc25-ce200479c4b5.png)

Removing the add-on and restarting FF showed a very different profile, and FF was up and responsive with a 3–4 seconds.

![image](https://user-images.githubusercontent.com/900542/221601748-abfd3cdc-3eb4-4c4e-abcc-3599516bb001.png)

I have about 450 tabs, of which 10 are 'active' on first load (9 pinned plus last 'normal' tab). The rest are discarded.

**Comment by SHHSSH** [comment]
> I have about 450 tabs, of which 10 are 'active' on first load (9 pinned plus last 'normal' tab). The rest are discarded.

Yeah, tbh, my last profile was pretty loaded with grouped tabs, I'd begun almost just utilising the whole of TST as a direct bookmarking feature in a sense. I began to realise, I had to discipline myself and backtrack to just curating some solid bookmarks & just quick-access bookmarking certain named folders that I can quickly open as a form of categorisation.

Then I just use TST's feature of re-opening bookmarks with structure intact.
Or, just long-term storage them with [%ANY("%GROUP%", "%TITLE% (+more)")% %DATE%/%MONTH%/%SHORT_YEAR% %HOURS%:%MINUTES% 
](https://github.com/piroor/treestyletab/issues/3209#issuecomment-1272619377)

So I dunno if it was just the loading of a massive tree structure of tabs, or a complex interaction with other components of FF.

**Comment by SHHSSH** [comment]
Additionally, I often wonder what the parameters are around TST not unloading/suspending pinned tabs when restarting FF. Perhaps this is out of bounds permission wise, I am not sure.

So I often thought that perhaps I was having unloading issues and that was factoring into the freezing of FF occasionally.

I'm currently choosing between https://github.com/pradeep-mishra/tab_suspender_firefox & TST's helper addon for unloading by a user @Lej77 - https://addons.mozilla.org/en-US/firefox/user/13464160/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_content=addons-manager-user-profile-link

I was using that helper addon for a while, though this one has become noticeably more lightweight, it seems... or maybe I am lying to myself.

I've always felt certain types of addons are more prone to memory malfunctions, e.g; dark modes, many of them often seem bloated beyond belief. I've recently found some serenity with - https://github.com/m-khvoinitsky/dark-background-light-text-extension.

**Comment by dsl101** [comment]
> Yeah, tbh, my last profile was pretty loaded with grouped tabs, I'd begun almost just utilising the whole of TST as a direct bookmarking feature in a sense. I began to realise, I had to discipline myself and backtrack to just curating some solid bookmarks & just quick-access bookmarking certain named folders that I can quickly open as a form of categorisation.
> 

That's exactly what I use TST for. Structured hierarchy of bookmarks, that are always in the same place, easily accessible, etc. I have the following addons installed, and no apparent memory issues:
* Bookmark Tree
* Move unloaded tabs
* Tab flip
* Tab Unloader
* MiddleClick

Plus my own (Tab Colour and soon Folder Expand Collapse). I haven't run the profiler on this add on, but looking at the source there is a loop over all tabs on first load to restore the 'locked' state, which includes an awaited async call to get the previous state. I suspect that's where the initial sluggishness is, and maybe an option to not persist the state (if you don't actually need that) would help here.

**Comment by jarredou** [comment]
So, I've finally fixed ALL the issues at the same time by uninstalling TST and all its addons, and installing SideBery instead. Tabs are not expanding if you don't ask them to, natively, no need to install unmaintained addons. Firefox is starting 20000x faster, uses less CPU, and way less RAM, no more crash each time sidebar is loaded.
Adieu.

**Comment by SHHSSH** [comment]
@jarredou I found that ALL my issues were being caused by Windscribe. Their addon is just simply not compatible with Firefox. It works as an OS VPN, a Chrome VPN, Linux etc. simple not Firefox, it was causing ALL my freezing.

Since then I have been able to have whatever addon I like and everything runs incredibly smooth. I have never wanted to defer from TST as I have optimised it so much so that it's the greatest addon imaginable for me. 

I am sorry to hear that for you though I genuinely believe it isn't*** TST. Unless there are some options you have set where you're loading all the tabs on startup, other than just pinned tabs which I believe automatically load.

**Comment by jarredou** [comment]
I don't have windscribe and never had it, and all the issues I'm facing since these TST's 2021 changes ...were from TST.

In my setup; only the current tab loads on startup. I don't use pinned tabs. The memory issue is only with this TST addon. When it's not activated, TST behave "normal" (but SideBery is way faster, it's crazy!)

One thing that is amazing is that all the tabs and all tree structure is conserved when you pass from TST to SideBery (I don't know if it works the other way, but if you want to test it, just make a copy of your profile, install it, and try it without the fear of data loss).

**Comment by SHHSSH** [comment]
> I don't have windscribe and never had it, and all the issues I'm facing since these TST's 2021 changes ...were from TST.
> 
> In my setup; only the current tab loads on startup. I don't use pinned tabs. The memory issue is only with this TST addon. When it's not activated, TST behave "normal" (but SideBery is way faster, it's crazy!)
> 
> One thing that is amazing is that all the tabs and all tree structure is conserved when you pass from TST to SideBery (I don't know if it works the other way, but if you want to test it, just make a copy of your profile, install it, and try it without the fear of data loss).

There's just too many customisations I've done for TST I have no interest in changing. I tried SideBerry at one point in time many years back when test running all sorts of Tree addons, glad you found your preference.
